lucille-core/src/main/java/com/kmwllc/lucille/stage/ComputeFieldSize.java

```diff
@@ -0,0 +1,45 @@
+package com.kmwllc.lucille.stage;
+
+import java.util.Iterator;
+import com.kmwllc.lucille.core.Document;
+import com.kmwllc.lucille.core.Stage;
+import com.kmwllc.lucille.core.StageException;
+import com.typesafe.config.Config;
+import com.kmwllc.lucille.core.*;
+
+/**
+ * Computes sizes of given byte array field and adds a field with that size
+ *
+ * Config Paramters:
+ *   - sourceField (String) : The field containing the byte array 
+ *   - outputField (String) : The name of the field which will be added to store the size. Will overwrite value if field already 
+ *                            exists
+ */
+public class ComputeFieldSize extends Stage {
+
+  private final String sourceField;
+  private final String outputField;
+
+  public ComputeFieldSize(Config config) {
+    super(config, new StageSpec().withRequiredProperties("sourceField", "outputField"));
+
+    this.sourceField = config.getString("sourceField");
```

Comment:
Other stages that take a source field generally refer to it as "source", not "sourceField". Let's try to be consistent with any implicit conventions like this. Should also check on "outputField" -- possibly rename to "destination"